Examples/ObjectiveC/POIAlongRouteExample.m

```diff
@@ -0,0 +1,140 @@
+    MGLLineStyleLayer *layer = [[MGLLineStyleLayer alloc] initWithIdentifier:@"polyline" source:source];
+    
+    // Set the line join and cap to a rounded end.
+    layer.lineJoin = [NSExpression expressionForConstantValue:[NSValue valueWithMGLLineJoin:MGLLineJoinRound]];
+    layer.lineCap = [NSExpression expressionForConstantValue:[NSValue valueWithMGLLineCap:MGLLineCapRound]];
+    
+    // Set the line color to a constant blue color.
+    layer.lineColor = [NSExpression expressionForConstantValue:[UIColor colorWithRed:59/255.0f green:178/255.0f blue:208/255.0f alpha:1]];
+    
+    // Use `NSExpression` to smoothly adjust the line width from 2pt to 20pt between zoom levels 14 and 18. The `interpolationBase` parameter allows the values to interpolate along an exponential curve
+    layer.lineWidth = [NSExpression expressionWithFormat:@"mgl_interpolate:withCurveType:parameters:stops:($zoomLevel, 'linear', nil, %@)",
+                       @{@14: @2, @18: @20}];
+    
+    // We can also add a second layer that will draw a stroke around the original line.
+    MGLLineStyleLayer *casingLayer = [[MGLLineStyleLayer alloc] initWithIdentifier:@"polyline-case" source:source];
+    // Copy these attributes from the main line layer.
+    casingLayer.lineJoin = layer.lineJoin;
+    casingLayer.lineCap = layer.lineCap;
+    // Line gap width represents the space before the outline begins, so should match the main line’s line width exactly.
+    casingLayer.lineGapWidth = layer.lineWidth;
+    // Stroke color slightly darker than the line color.
+    casingLayer.lineColor = [NSExpression expressionForConstantValue:[UIColor colorWithRed:41/255.0f green:145/255.0f blue:171/255.0f alpha:1]];
+    // Use a style function to gradually increase the stroke width between zoom levels 14 and 18.
+    casingLayer.lineWidth = [NSExpression expressionWithFormat:@"mgl_interpolate:withCurveType:parameters:stops:($zoomLevel, 'linear', nil, %@)",
+                             @{@14: @1, @18: @4}];
+
+    // Just for fun, let’s add another copy of the line with a dash pattern.
+    MGLLineStyleLayer *dashedLayer = [[MGLLineStyleLayer alloc] initWithIdentifier:@"polyline-dash" source:source];
+    dashedLayer.lineJoin = layer.lineJoin;
+    dashedLayer.lineCap = layer.lineCap;
+    dashedLayer.lineWidth = layer.lineWidth;
+    dashedLayer.lineColor = [NSExpression expressionForConstantValue:[UIColor whiteColor]];
+    dashedLayer.lineOpacity = [NSExpression expressionForConstantValue:@0.5];
+    // Dash pattern in the format [dash, gap, dash, gap, ...]. You’ll want to adjust these values based on the line cap style.
+    dashedLayer.lineDashPattern = [NSExpression expressionForConstantValue:@[@0, @1.5]];
+    
+    MGLSymbolStyleLayer *poiLayer = (MGLSymbolStyleLayer *)[self.mapView.style layerWithIdentifier:@"poi-label"];
+    [self.mapView.style insertLayer:layer belowLayer:poiLayer];
+    [self.mapView.style insertLayer:dashedLayer belowLayer:layer];
```

Comment:
This is currently adding the dashed layer below the main layer so it is not visible. I'd recommend switching it to `aboveLayer` so it is visible and matches the Swift example.